test/avalanche/token_cache_test.exs

```diff
@@ -37,9 +37,11 @@ defmodule Avalanche.TokenCacheTest do
   -----END PRIVATE KEY-----
   """
 
+  setup :set_mimic_global
```

Comment:
do we need this one? We are about to move away from Mimic. If we do, no worries and we will tackle this later

Reply:
I think this was for the flaky tests. Let's leave it and remove when we remove Mimic.

Reply:
Yes, mimic doesn't work great across process boundaries, which it would be doing here in the call to the token cache